activestorage/lib/active_storage/engine.rb

```diff
@@ -34,8 +34,8 @@ class Engine < Rails::Engine # :nodoc:
     end
 
     initializer "active_storage.services" do
-      config.after_initialize do |app|
-        if config_choice = app.config.active_storage.service
+      config.to_prepare do
+        if config_choice = Rails.configuration.active_storage.service
```

Comment:
Instead of reading the YAML every request in development what do you think to store the configuration in the `active_storage` configuration object? Something like:

```
initializer "active_storage.services" do
  config.active_storage.service_configuration = application.config_for("config/storage.yml")
  
  config.to_prepare do
    if config_choice = Rails.configuration.active_storage.service
      ActiveStorage::Blob.service =
             begin
               ActiveStorage::Service.configure config_choice, Rails.configuration.active_storage.service_configuration
             rescue => e
               raise e, "Cannot load `Rails.config.active_storage.service`:\n#{e.message}", e.backtrace
             end
     endddd